client/my-sites/posts/post-controls.jsx

```diff
@@ -1,244 +1,223 @@
-		onHideMore: React.PropTypes.func.isRequired,
-		onPublish: React.PropTypes.func,
-		onTrash: React.PropTypes.func,
-		onDelete: React.PropTypes.func,
-		onRestore: React.PropTypes.func,
-	},
-
-	view: function() {
-		recordEvent( 'Posts', 'Clicked View Post' );
-	},
-
-	preview: function() {
-		recordEvent( 'Posts', 'Clicked Preview Post' );
-	},
-
-	edit: function() {
-		recordEvent( 'Posts', 'Clicked Edit Post' );
-	},
-
-	copy: function() {
-		recordEvent( 'Posts', 'Clicked Copy Post' );
-	},
-
-	viewStats: function() {
-		recordEvent( 'Posts', 'Clicked View Post Stats' );
-	},
-
-	buildControls: function( controlsArray ) {
-		var postId = this.props.post.ID,
-			controls;
-
-		controls = controlsArray.map( function( item, i ) {
-			const itemKey = 'controls-' + postId + '-' + i;
-			const disabled = item.disabled || false;
-
-			return (
-				<li className={ classNames( { 'post-controls__disabled': disabled } ) } key={ itemKey }>
-					<a href={ item.href } className={ item.className } onClick={ disabled ? noop : item.onClick } target={ item.target ? item.target : null }>
-						<Gridicon icon={ item.icon } size={ 18 } />
-						<span>{ item.text }</span>
-					</a>
-				</li>
-			);
-		}, this );
-
-		return controls;
-	},
-
-	render: function() {
-		var post = this.props.post,
-			availableControls = [],
-			extraControls = [],
-			parsed, previewURL, statsURL, postControlsClass, mainControls, moreControls, i, moreControlsSpliceIndex;
-
-		// NOTE: Currently Jetpack site posts are not returning `post.capabilities`
-		// and those posts will not have access to post management type controls
-
-		// --- Main Controls (not behind ... more link) ---
-		if ( utils.userCan( 'edit_post', post ) ) {
-			availableControls.push( {
-				text: this.translate( 'Edit' ),
-				className: 'post-controls__edit',
-				href: this.props.editURL,
-				target: null,
-				onClick: this.edit,
-				icon: 'pencil'
+import { isEnabled } from 'config';
+import { ga } from 'lib/analytics';
+import { userCan } from 'lib/posts/utils';
+import Gridicon from 'components/gridicon';
+
+const view = () => ga.recordEvent( 'Posts', 'Clicked View Post' );
+const preview = () => ga.recordEvent( 'Posts', 'Clicked Preiew Post' );
+const edit = () => ga.recordEvent( 'Posts', 'Clicked Edit Post' );
+const copy = () => ga.recordEvent( 'Posts', 'Clicked Copy Post' );
+const viewStats = () => ga.recordEvent( 'Posts', 'Clicked View Post Stats' );
+
+const getAvailableControls = props => {
+	const {
+		editURL,
+		fullWidth,
+		onDelete,
+		onHideMore,
+		onPublish,
+		onRestore,
+		onShowMore,
+		onToggleShare,
+		onTrash,
+		post,
+		site,
+		translate,
+	} = props;
+	const controls = { main: [], more: [] };
+
+	// NOTE: Currently Jetpack site posts are not returning `post.capabilities`
+	// and those posts will not have access to post management type controls
+
+	// Main Controls (not behind ... more link)
+	if ( userCan( 'edit_post', post ) ) {
```

Comment:
Also note that if this component is connected in the future, we can use the following in mapStateToProps
```
import { canCurrentUser } from 'state/current-user/selectors';
//...
const userCanEditPost = canCurrentUser( state, siteId, 'edit_post');
```